evalml/model_understanding/prediction_explanations/_algorithms.py

```diff
@@ -97,10 +99,12 @@ def _compute_shap_values(pipeline, features, training_data=None):
         mappings = []
         for class_shap_values in shap_values:
             mappings.append(_create_dictionary(class_shap_values, feature_names))
-        return mappings
+        return (mappings, explainer) if return_explainer else mappings
     # regression problem
     elif isinstance(shap_values, np.ndarray):
-        return _create_dictionary(shap_values, feature_names)
+        dic = _create_dictionary(shap_values, feature_names)
+        return (dic, explainer) if return_explainer else dic
```

Comment:
Returning variable length arguments can make consuming output a bit uglier up the stack. I suggest we always return a tuple of two arguments. If `explainer` here happens to return as `None`, that's fine.

Rolling with @freddyaboulton 's suggestion above though, I wonder if we can just delete `return_explainer` and simply always return both the dictionary and the explainer here.

Reply:
Done, this is now returning the expected_value instead of the explainer.  Always.